src/common.h

```diff
@@ -760,6 +760,13 @@ enum
 
 	SFE_OPUS_BAD_SAMPLERATE,
 
+	SFE_WAVPACK_NEW_DECODER,
+	SFE_WAVPACK_NEW_ENCODER,
+	SFE_WAVPACK_WRITE_HEADER,
+	SFE_WAVPACK_PACK_SAMPLES,
+	SFE_WAVPACK_UNPACK_SAMPLES,
```

Comment:
Error type should describe *why* there is an error. These types describe *where* there is an error.

In most cases, try and return a generic error, based on the error value from the codec library, such as SFE_MALFORMED_FILE or SFE_UNSUPPORTED. It is okay to add format specific errors, but they should provide format-specific insight of why there is an error.

(Yes, I know, `SFE_FLAC_NEW_DECODER` exists, but the Flac code is the least libsndfile-like code in the tree. :-P )

Format specific error types should only be added if there is an error that is specific to that format (eg, mono only, bad samplerate, format specific construct is corrupt).